### DISCUSSION TITLE
"use client" for components within page directory?

### DISCUSSION THREAD

**Discussion by Abyrd9** [vercel/next.js#45769] [Help]
Hello,

I'm getting the error for "use-client" from a component that's using some [floating-ui](https://floating-ui.com/) components.
`useInsertionEffect only works in Client Components. Add the "use client" directive at the top of the file to use it.`

I am on next version `13.1.6`, but still am only using the pages directory. I thought that use-client would only work for components nested under the app directory, is that not true? I added "use client" to the top of the component, but still receiving the error. Trying to figure out if this a bug or not.

**Comment by ciruz** [comment]
Hm, this sounds like there is something wrong with floating-ui.

**Comment by dizlexik** [comment]
FWIW, I received this exact same error, also originating from within floating-ui, and after much troubleshooting the solution ended up being unrelated and I had to set the `neverSuspend` option to true in `withUrqlClient`. So maybe that's something to look at if you're also using `next-urql`, or maybe some other library that's trying to do suspense magic.

**Comment by isoaxe** [comment]
I got this error when using the Mantine UI library. See discussion any my [comment](https://github.com/mantinedev/mantine/issues/2880#issuecomment-1491456551).
